### PR TITLE
Add liquidity

### PR DESCRIPTION
**Pull request by jsidorenko**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "axios": "1.3.4",
     "bootstrap": "5.2.3",
     "busboy": "1.6.0",
+    "decimal.js": "^10.4.3",
```
Comment (VadimSaveljev) [inline package.json]: strict version please

File: src/components/Common/NotFound.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { memo } from 'react';
+
```
Comment (VadimSaveljev) [inline src/components/Common/NotFound.tsx:2]: add a TODO, will need to tell designer to create a 404 template

File: src/components/Pages/Assets/Pools/AddLiquidity.tsx [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+} from '@helpers/utilities';
+
+import { useAssets } from '@hooks/useAssets';
+
+interface AddLiquidityProps {
+  asset1: PalletAssetConversionMultiAssetId;
+  asset2: PalletAssetConversionMultiAssetId;
+}
+
+const getCleanFormattedBalance = (planck: BN, decimals: number): string => {
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Pools/AddLiquidity.tsx]: will this be used anywhere else potentially? can go to helpers if so

File: src/components/Pages/Assets/Pools/AddLiquidity.tsx [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+      }
+
+      if (formError) {
+        setStatus({ type: ModalStatusTypes.ERROR, message: formError });
+        openModalStatus();
+        return;
+      }
+
+      // add slippage tolerance to min amounts
+      const slippage = 0.5; // 0.5%
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Pools/AddLiquidity.tsx]: do we have any standards for these?

File: src/components/Pages/Assets/Pools/AddLiquidity.tsx [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    }
+  }, [api, getNativeMetadata, nativeMetadata]);
+
+  useEffect(() => {
+    if (poolReserves && !isPoolEmpty(poolReserves) && assetMetadata && nativeMetadata) {
+      const asset1Reserve = getCleanFormattedBalance(poolReserves[0], nativeMetadata.decimals);
+      const asset2Reserve = getCleanFormattedBalance(poolReserves[1], getAssetDecimals(assetMetadata[0]));
+      setExchangeRate(calcExchangeRate(+asset1Reserve, +asset2Reserve));
+    }
+  }, [poolReserves, assetMetadata, nativeMetadata]);
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Pools/AddLiquidity.tsx]: this is too much
should add TODO to refactor gathering data from when user starts to do stuff

File: src/components/Pages/Assets/Pools/AddLiquidity.tsx (hunk 4 above)
Comment (jsidorenko) [reply src/components/Pages/Assets/Pools/AddLiquidity.tsx]: Nope, just best practises

**Review by VadimSaveljev — APPROVED** [review]

**Pull request merged**